test/Microsoft.AspNetCore.Http.Connections.Tests/NegotiateProtocolTests.cs

```diff
@@ -55,6 +55,27 @@ public void ParsingNegotiateResponseMessageThrowsForInvalid(string payload, stri
             Assert.Equal(expectedMessage, exception.InnerException.Message);
         }
 
+        [Fact]
+        public void ParsingClassicSignalRResponseThrowsError()
+        {
+            var payload = "{\"Url\":\"/signalr\"," +
+                "\"ConnectionToken\":\"X97dw3uxW4NPPggQsYVcNcyQcuz4w2\"," +
+                "\"ConnectionId\":\"05265228-1e2c-46c5-82a1-6a5bcc3f0143\"," +
+                "\"KeepAliveTimeout\":10.0," +
+                "\"DisconnectTimeout\":5.0," +
+                "\"TryWebSockets\":true," +
+                "\"ProtocolVersion\":\"1.5\"," +
+                "\"TransportConnectTimeout\":30.0," +
+                "\"LongPollDelay\":0.0}";
+
+            var responseData = Encoding.UTF8.GetBytes(payload);
+            var ms = new MemoryStream(responseData);
+
+            var exception = Assert.Throws<InvalidDataException>(() => NegotiateProtocol.ParseResponse(ms));
+
+            Assert.Equal("It seems like you are trying to connect to an ASP.NET SignalR Server. This client only supports connecting to an ASP.NET Core SignalR Server.", exception.InnerException.Message);
```

Comment:
seems? LOL. What is this clippy? "It looks like"

Reply:
Because I want word-smithing! Give me suggestions!